src/Http/Requests/Admin/SettingsUpdateRequest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace EscolaLms\Settings\Http\Requests\Admin;
+
+use EscolaLms\Settings\Models\Setting;
+use EscolaLms\Settings\Enums\SettingTypes;
+use Illuminate\Validation\Rule;
+use Illuminate\Foundation\Http\FormRequest;
+
+class SettingsUpdateRequest extends FormRequest
+{
+    public function authorize()
+    {
+        return $this->user()->can('delete', Setting::class);
+    }
+
+    public function rules()
+    {
+        return [
+            'key' => ['sometimes', 'string'],
+            'group' => ['sometimes', 'string'],
+            'public' => ['boolean'],
+            'enumerable' => ['boolean'],
+            'sort' => ['integer'],
+            'type' => ['sometimes',  Rule::in(SettingTypes::getValues())],
```

Comment:
I wonder if we want `type` to be changeable during update? Because different types have uncompatible values...

Reply:
This is pure admin only feature, so I'm not designing it as bullet/idiot proof :)